plugins/http_plugin/beast_http_session.hpp

```diff
@@ -0,0 +1,587 @@
+                                static_cast<int>(http::status::not_found)); 
+                };
+                
+                // Request path must be absolute and not contain "..".
+                if( req.target().empty() ||
+                    req.target()[0] != '/' ||
+                    req.target().find("..") != beast::string_view::npos)
+                    return bad_request("Illegal request-target", *this);
+
+                try {
+                    if(!derived().allow_host(req))
+                        return;
+
+                    if( !plugin_state_->access_control_allow_origin.empty()) {
+                        res.set( "Access-Control-Allow-Origin", plugin_state_->access_control_allow_origin );
+                    }
+                    if( !plugin_state_->access_control_allow_headers.empty()) {
+                        res.set( "Access-Control-Allow-Headers", plugin_state_->access_control_allow_headers );
+                    }
+                    if( !plugin_state_->access_control_max_age.empty()) {
+                        res.set( "Access-Control-Max-Age", plugin_state_->access_control_max_age );
+                    }
+                    if( plugin_state_->access_control_allow_credentials ) {
+                        res.set( "Access-Control-Allow-Credentials", "true" );
+                    }
+
+                    // Respond to options request
+                    if(req.method() == http::verb::options)
+                    {
+                        send_response("", static_cast<int>(http::status::ok));
+                        return;
+                    }
+
+                    // verfiy bytes in flight/requests in flight
+                    if( !verify_max_bytes_in_flight() ) return;
+
+                    std::string resource = std::string(req.target());
+                    // look for the URL handler to handle this reosouce
+                    auto handler_itr = plugin_state_->url_handlers.find( resource );
+                    if( handler_itr != plugin_state_->url_handlers.end()) {
+                        std::string body = req.body();
+                        handler_itr->second( derived().shared_from_this(), 
+                                            std::move( resource ), 
+                                            std::move( body ), 
+                                            make_http_response_handler(thread_pool_, plugin_state_, derived().shared_from_this()) );
+                    } else {
+                        fc_dlog( logger, "404 - not found: ${ep}", ("ep", resource) );
+                        not_found(resource, *this);                    
+                    }
+                } catch (...) {
+                    handle_exception();
+                }
+            }
+
+            void report_429_error(const std::string & what) {                
+                send_response(std::string(what), 
```

Comment:
Extra copy of `what` not needed.

Reply:
`report_429_error` could take a `std::string` which could then be moved into `send_response`